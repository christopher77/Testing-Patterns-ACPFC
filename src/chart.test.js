import React from "react";
import { render } from "@testing-library/react";
import Chart from "./chart";

test("Render the new chart", () => {
  const employees = [
    {
      id: "Fri Jun 07 2019 17:08:11 GMT-0500 (Peru Standard Time)",
      name: "CEO",
      title: "CEO",
      parentId: null
    },
    {
      id: "Fri Jun 07 2019 17:08:16 GMT-0500 (Peru Standard Time)",
      name: "Carlos",
      title: "Dev",
      parentId: "Fri Jun 07 2019 17:08:11 GMT-0500 (Peru Standard Time)"
    },
    {
      id: "Fri Jun 07 2019 17:08:21 GMT-0500 (Peru Standard Time)",
      name: "Daniel",
      title: "Dev",
      parentId: "Fri Jun 07 2019 17:08:16 GMT-0500 (Peru Standard Time)"
    }
  ];

  const handleCreateChild = jest.fn();
  const { asFragment } = render(
    <Chart employees={employees} handleCreateChild={handleCreateChild} />
  );
  expect(asFragment()).toMatchSnapshot();
});
