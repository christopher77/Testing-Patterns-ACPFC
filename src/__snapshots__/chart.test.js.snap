// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the new chart 1`] = `
<DocumentFragment>
  <div
    class="tree"
  >
    <ul
      class="css-1rjgz6v-Chart"
    >
      <li
        class="css-cqjg2n-Employee"
      >
        <div
          class="css-98qhp7-Employee"
        >
          CEO - CEO
          <form>
            <input
              aria-label="Employee name"
              id="employee-name"
              placeholder="Employee name"
              type="text"
              value=""
            />
          </form>
        </div>
        <ul
          class="css-jneiqc-Employee"
        >
          <li
            class="css-cqjg2n-Employee"
          >
            <div
              class="css-98qhp7-Employee"
            >
              Carlos - Dev
              <form>
                <input
                  aria-label="Employee name"
                  id="employee-name"
                  placeholder="Employee name"
                  type="text"
                  value=""
                />
              </form>
            </div>
            <ul
              class="css-jneiqc-Employee"
            >
              <li
                class="css-cqjg2n-Employee"
              >
                <div
                  class="css-98qhp7-Employee"
                >
                  Daniel - Dev
                  <form>
                    <input
                      aria-label="Employee name"
                      id="employee-name"
                      placeholder="Employee name"
                      type="text"
                      value=""
                    />
                  </form>
                </div>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
